constructor/fcp.py

```diff
@@ -176,9 +180,49 @@ def check_duplicates_files(pc_recs, platform, ignore_duplicate_files=False):
     return total_tarball_size, total_extracted_pkgs_size
 
 
+def _precs_from_environment(environment, download_dir, conda_exe="conda.exe"):
+    from subprocess import check_output
+
+    # get basic data about the environment's packages
+    list_flag = "--prefix" if isdir(environment) else "--name"
```

Comment:
I'm finding this isn't working for me in `--name` mode, and I think it's because the standalone conda executable has a different envs_dirs path than the surrounding conda environment. Does that make sense?

Here's what I did: I created a test environment called `ctest`, with the prefix `/Users/mgrant/miniconda3/envs/ctest`, and I added `environment: ctest` to my `construct.yaml`. Then running `constructor .` crashed with
```
subprocess.CalledProcessError: Command '['/Users/mgrant/miniconda3/standalone_conda/conda.exe', 'list', '--name', 'ctest', '--json']' returned non-zero exit status 1
```
When I try running that command, I get this:
```
{
  "caused_by": "None",
  "error": "EnvironmentLocationNotFound: Not a conda environment: /Users/mgrant/.conda/envs/ctest",
  "exception_name": "EnvironmentLocationNotFound",
  "exception_type": "<class 'conda.exceptions.EnvironmentLocationNotFound'>",
  "location": "/Users/mgrant/.conda/envs/ctest",
  "message": "Not a conda environment: /Users/mgrant/.conda/envs/ctest"
}
```
If I change my `environment:` value to a full path, it works fine. The solution, it seems to me, is either to consider using `os.environ['CONDA_EXE']` instead of `conda_exe`, or to override `CONDA_ENVS_DIRS` temporarily with the appropriate value so it sees the environment directory.

Reply:
It makes sense when I look at the output of `/Users/mgrant/miniconda3/standalone_conda/conda.exe info`:
```
       envs directories : /Users/mgrant/.conda/envs
                          /var/folders/zr/v087ks_x1dz1vr46g7jqb1880000gp/T/_MEIuACwao/envs
```

Reply:
The more I think about this, the more I believe that your _environment_ `conda` needs to be used to render the environment, no matter what the version of `conda.exe` is. After all, it seems to me that presumably you are interactively designing and adjusting this environment until it is precisely what you want to see. So the idiosyncracies of your local conda installation are exactly what you want to capture. If for any reason at all the render produced by constructor differs from your local conda, you're likely to prefer the latter.

Reply:
> I'm finding this isn't working for me in --name mode, and I think it's because the standalone conda executable has a different envs_dirs path than the surrounding conda environment. Does that make sense? 

Yeah that makes sense to me and is what I found as well. 

> your environment conda needs to be used to render the environment

I think this makes sense. I'll make the adjustments